### PR TITLE
Run Chip Tool without sudo

### LINKED ISSUE #17: Make chip-tool command work without root privileges
The `chip-tool` command installed by the snap requires root privileges to function. This is because the storage and config files are written under [`/mnt`](https://github.com/canonical/chip-tool-snap/blob/82b6c38a46cfec77b65fac0a611161b0eccef8b0/snap/snapcraft.yaml#L89-L92), which is bind mounted on [`/var/snap/chip-tool/common/mnt/`](https://github.com/canonical/chip-tool-snap/blob/82b6c38a46cfec77b65fac0a611161b0eccef8b0/snap/snapcraft.yaml#L17-L18). This directory is owned by the root user.

This also makes it problematic to use the tool independently with different users. 

Ideally, the storage and config paths should be user specific (e.g. read from [`SNAP_USER_COMMON`](https://snapcraft.io/docs/environment-variables#heading--snap-user-common)). This is currently not easily achievable since the paths are set at compile time.

### PR DESCRIPTION
**Pull request by jpm-canonical**
We had to use sudo because of the directory where Chip Tool stores its config and data (Resolves #17). After an upstream changes we can set the `TMPDIR` env var to tell Chip Tool where to store its data.

This PR tries to:
* Set `TMPDIR` to snap home dir (`SNAP_USER_COMMON`) using command chain
* Removed outdated path rewrites and unnecessary mounts

⚠️ Testing is needed to see if this change is actually effective, and previous functionality is unchanged.

Log output of Chip Tool on where exactly data is stored:
```
$ chip-tool onoff toggle 110 1
[1714639755.999521][286634:286634] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /home/jpmeijers/snap/chip-tool/common/chip_tool_kvs
[1714639755.999830][286634:286634] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_kvs
[1714639755.999840][286634:286634] CHIP:DL: ChipLinuxStorage::Init: Attempt to re-initialize with KVS config file: /tmp/chip_kvs
[1714639756.001540][286634:286634] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_factory.ini
[1714639756.001573][286634:286634] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_config.ini
[1714639756.001588][286634:286634] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_counters.ini
[1714639756.001656][286634:286634] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-mbO1FA)
[1714639756.001757][286634:286634] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
...
[1714639756.260402][286634:286634] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-OSmUHU)
[1714639756.260762][286634:286634] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1714639756.260837][286634:286634] CHIP:DL: NVS set: chip-counters/total-operational-hours = 0 (0x0)
```

This looks like `chip_tool_*` files are stored in the correct persistent directory, but `chip_*` files are still stored under `/tmp`.

### CODE REVIEW

**Comment by jpm-canonical** [comment]
Previously there was an issue where history is not persisted (#20).

Using this branch, running a few commands in interactive mode, we can see the history is now stored in `$SNAP_USER_COMMON`.

```
$ chip-tool interactive start
>>> onoff toggle 110 1
[1714638608.210295][285852:285852] CHIP:TOO: Command: onoff toggle 110 1
>>> quit
[1714638614.446034][285852:285852] CHIP:TOO: Command: quit 
[1714638614.446330][285852:285852] CHIP:TOO: Unknown cluster or command set: quit
>>> exit
[1714638616.938403][285852:285852] CHIP:TOO: Command: exit 
[1714638616.938520][285852:285852] CHIP:TOO: Unknown cluster or command set: exit
^C
$ cd /home/jpmeijers/snap/chip-tool/common
$ ls
chip_tool_config.alpha.ini  chip_tool_history
chip_tool_config.ini        chip_tool_kvs
$ cat chip_tool_history 
onoff toggle 110 1
quit
exit
```

**Comment by jpm-canonical** [comment]
❌ If `chip-tool` was run with sudo, and you try to run it without sudo after that, it can't open the files in `/tmp`, as they are owned by root.

```
$ chip-tool pairing onnetwork 110 20202021
[1714744727.897352][25324:25324] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /home/jpmeijers/snap/chip-tool/common/chip_tool_kvs
[1714744727.897477][25324:25324] CHIP:DL: writing settings to file (/home/jpmeijers/snap/chip-tool/common/chip_tool_kvs-LoTsJE)
[1714744727.897558][25324:25324] CHIP:DL: renamed tmp file to file (/home/jpmeijers/snap/chip-tool/common/chip_tool_kvs)
[1714744727.897986][25324:25324] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_kvs
[1714744727.897996][25324:25324] CHIP:DL: ChipLinuxStorage::Init: Attempt to re-initialize with KVS config file: /tmp/chip_kvs
[1714744727.900503][25324:25324] CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_factory.ini
[1714744727.900593][25324:25324] CHIP:DL: writing settings to file (/tmp/chip_factory.ini-qTVr56)
[1714744727.900640][25324:25324] CHIP:DL: failed to rename (/tmp/chip_factory.ini-qTVr56), Operation not permitted (1)
[1714744727.900659][25324:25324] CHIP:DL: Configuration Manager initialization failed: src/platform/Linux/CHIPLinuxStorageIni.cpp:116: CHIP Error 0x000000AF: Write to file failed
[1714744727.900669][25324:25324] CHIP:-: src/platform/Linux/CHIPLinuxStorageIni.cpp:116: CHIP Error 0x000000AF: Write to file failed at ../examples/chip-tool/commands/common/CHIPCommand.cpp:142
[1714744727.900679][25324:25324] CHIP:TOO: Run command failure: src/platform/Linux/CHIPLinuxStorageIni.cpp:116: CHIP Error 0x000000AF: Write to file failed
[1714744727.912484][25324:25324] CHIP:SPT: VerifyOrDie failure at src/system/SystemLayerImplSelect.h:59: mLayerState.Destroy()
Aborted (core dumped)
```

File: tests/common.go [hunk 1]
```diff
@@ -31,3 +38,27 @@ func InstallChipTool(t *testing.T) {
...
+			if t != nil {
+				t.Logf("%s installed version %s\n", snapName, columns[1])
+			} else {
+				log.Printf("%s installed version %s\n", snapName, columns[1])
+			}
+			return nil
+		}
+	}
+
+	return errors.New(fmt.Sprintf("snap '%s' not found", snapName))
```
Comment (farshidtz) [inline tests/common.go:63]: Could use https://pkg.go.dev/github.com/canonical/matter-snap-testing@v1.0.0-beta.3/utils#SnapVersion?

File: tests/upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	require.NoError(t,
+		utils.SnapInstallFromFile(nil, utils.LocalServiceSnapPath),
+	)
+
+	// Control device again
+	t.Run("Control Local", func(t *testing.T) {
+		err := PrintSnapVersion(t, chipToolSnap)
+		if err != nil {
+			t.Fatalf("snap not installed")
+		}
```
Comment (farshidtz) [inline tests/upgrade_test.go:93]: Use [`assert.NoError`](https://pkg.go.dev/github.com/stretchr/testify/assert#NoError) and pass a custom error message.

**Review by farshidtz — CHANGES_REQUESTED** [review]
Thanks. It looks good apart from a few minor issues.

File: tests/common.go [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
+	"log"
+	"regexp"
+	"strings"
 	"testing"
 
 	"github.com/canonical/matter-snap-testing/utils"
 	"github.com/stretchr/testify/require"
 )
 
+const allClusterSnap = "matter-all-clusters-app"
```
[the remark this answers is not in the document]
Comment (jpm-canonical) [reply tests/common.go:10]: This typo was there originally. Let's fix it in a separate PR to keep this one's diff clean.

[the remark this answers is not in the document]
Comment (jpm-canonical) [reply tests/common.go:10]: https://github.com/canonical/chip-tool-snap/issues/64

**Review by farshidtz — APPROVED** [review]
Great. Thanks a lot for the fix and adding tests.

**Pull request merged**